### PR TITLE
Mailinglist

### PR DESCRIPTION
**Pull request by xdamman**

### CODE REVIEW

File: server/controllers/services/email.js [hunk 1]
```diff
@@ -207,7 +220,30 @@ module.exports = (app) => {
         return Promise.map(users, (user) => emailLib.send('email.approve', `members@${slug}.opencollective.com`, getData(user), {bcc:user.email}));
       })
       .then(() => res.send('Mailgun webhook processed successfully'))
-      .catch(next);
+      .catch(e => {
+        switch (e.message) {
+          case 'no_subscribers':
```
Comment (sedubois) [inline server/controllers/services/email.js]: these constants should be extracted

File: test/email.routes.test.js [hunk 2]
```diff
@@ -127,6 +129,20 @@ describe("email.routes.test", () => {
       });
   });
 
+  it("rejects emails sent to unknown mailing list", (done) => {
+
+    const unknownMailingListWebhook = _.defaults({ recipient: 'unknown@testcollective.opencollective.com' }, webhookBody);
```
Comment (sedubois) [inline test/email.routes.test.js]: actually I guess it's the same to use `Object.assign` (no need for external lib)?

File: test/email.routes.test.js [hunk 3]
```diff
@@ -127,6 +129,20 @@ describe("email.routes.test", () => {
       });
   });
 
+  it("rejects emails sent to unknown mailing list", (done) => {
```
Comment (sedubois) [inline test/email.routes.test.js]: no need for `done`, mocha handles promises:

```
it(..., () => request(app)
  .post(...)
  .send(...)
  .then(res => {
    expect(...);
    ...
  });
```

File: server/controllers/services/email.js (hunk 1 above)
Comment (xdamman) [inline server/controllers/services/email.js]: there is no point if they are only used once.

File: test/email.routes.test.js (hunk 2 above)
Comment (xdamman) [inline test/email.routes.test.js]: You are right! I'll change it.

File: test/email.routes.test.js (hunk 3 above)
Comment (xdamman) [inline test/email.routes.test.js]: I didn't know. Ok, changed.

File: server/controllers/services/email.js (hunk 1 above)
Comment (sedubois) [inline server/controllers/services/email.js]: But they are used twice, e.g lines 194 + 225. If you change one without the other, code breaks.

Comment (xdamman) [inline server/controllers/services/email.js]: but it's only in one file. Honestly, in the age of modern IDE and autocomplete, it doesn't really matter if it's in one single file.

**Pull request merged**